server/api/impl/admin/synchronize.go

```diff
@@ -0,0 +1,227 @@
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+const oldVersionKey = "update_from_version"
+const appsJSONFile = "apps.json"
+const callOnceKey = "PP_CallOnce_key"
+
+// AppVersions describes versions for all the apps in all installations
+type AppVersions struct {
+	Apps      map[apps.AppID]string            `json:"apps"`
+	Overrides map[string]map[apps.AppID]string `json:"overrides"`
+}
+
+func getAppsForInstallation(installationID string) (map[apps.AppID]string, error) {
+	data, err := ioutil.ReadFile(fmt.Sprintf("assets/%s", appsJSONFile))
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't read %s file", appsJSONFile)
+	}
+	var allAppVersions *AppVersions
+	if err := json.Unmarshal(data, &allAppVersions); err != nil || allAppVersions == nil {
+		return nil, errors.Wrapf(err, "can't unmarshal %s file", appsJSONFile)
+	}
+
+	apps := allAppVersions.Apps
+	if overrides, ok := allAppVersions.Overrides[installationID]; ok {
+		for id, version := range overrides {
+			apps[id] = version
+		}
+	}
+	return apps, nil
+}
+
+func (adm *Admin) populateManifests(appVersions map[apps.AppID]string) {
+	adm.store.Manifest().Cleanup()
+	for id, version := range appVersions {
+		manifest, err := adm.awsClient.GetManifest(id, version)
+		if err != nil {
+			// Note that we are not returning an error here.
+			adm.mm.Log.Error("can't get manifest for", "app", id, "version", version, "err", err)
+			continue
+		}
+		adm.store.Manifest().Save(manifest)
+	}
+}
+
+// LoadAppsList synchronizes apps with the apps.json file.
+func (adm *Admin) LoadAppsList() error {
+	installationID := adm.mm.System.GetDiagnosticID()
+	appsForInstallation, err := getAppsForInstallation(installationID)
+	if err != nil {
+		return errors.Wrap(err, "can't get apps for installation")
+	}
+
+	adm.populateManifests(appsForInstallation)
+
+	registeredApps := adm.store.App().GetAll()
+
+	registeredAppsMap := map[apps.AppID]*apps.App{}
+	updatedAppVersionsMap := map[apps.AppID]string{}
+	// Update apps
+	for _, registeredApp := range registeredApps {
+		registeredAppsMap[registeredApp.Manifest.AppID] = registeredApp
+		manifest, err := adm.store.Manifest().Get(registeredApp.Manifest.AppID)
+		if err != nil {
+			return errors.Wrapf(err, "can't load manifest from store appID = %s", registeredApp.Manifest.AppID)
+		}
+		if err := adm.UninstallApp(registeredApp); err != nil {
+			return errors.Wrapf(err, "can't delete an app")
+		}
+		if manifest.Version != registeredApp.Manifest.Version {
+			// update app in proxy plugin
+			oldVersion := registeredApp.Manifest.Version
+			registeredApp.Manifest = manifest
+			if err := adm.store.App().Save(registeredApp); err != nil {
+				return errors.Wrapf(err, "can't store app - %s", registeredApp.Manifest.AppID)
+			}
+			updatedAppVersionsMap[registeredApp.Manifest.AppID] = oldVersion
+		}
+	}
+
+	// Add new apps as registered
```

Comment:
since we cannot add them as smth. else (other than registered), maybe we can just say "register new apps"